clap_complete/src/dynamic/shells/mod.rs

```diff
@@ -16,6 +16,79 @@ use crate::dynamic::Completer as _;
 /// A subcommand definition to `flatten` into your CLI
 ///
 /// This provides a one-stop solution for integrating completions into your CLI
+///
+/// # Examples
+///
+/// The following example shows how to integrate completions into your CLI and generate completions.
+///
+/// 1. Build an application with derive API.
+/// 2. Call `clap_complete::dynamic::shells::CompleteCommand::augment_subcommands` to add the `complete` subcommand into the application.
+/// 3. Call `get_matches()`, or any of the other normal methods directly after.
+///
+/// For example:
+///
+/// ```no_run
+/// // src/main.rs
+/// use clap::{CommandFactory, FromArgMatches, Parser, Subcommand};
+///
+/// #[derive(Parser, Debug)]
+/// #[clap(name = "dynamic", about = "A dynamic command line tool")]
+/// struct Cli {
+///     /// Input file path
+///     #[clap(short, long, value_hint = clap::ValueHint::FilePath)]
+///     input: Option<String>,
+
+///     /// Output format
+///     #[clap(short = 'F', long, value_parser = ["json", "yaml", "toml"])]
+///     format: Option<String>,
+/// }
+///
+/// fn main() {
+/// 	let cmd = Cli::command();
+/// 	let cmd = clap_complete::dynamic::shells::CompleteCommand::augment_subcommands(cmd);
+/// 	let matches = cmd.get_matches();
+/// 	if let Ok(completions) =
+/// 		clap_complete::dynamic::shells::CompleteCommand::from_arg_matches(&matches)
+/// 	{
+/// 		completions.complete(&mut Cli::command());
+/// 	}
+///
+/// 	// normal logic continues...
```

Comment:
This is not a common pattern for the derive API.  The user should instead `#[command(flatten)]` it into their own CLI